Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -1656,4 +1656,17 @@ class HTTPClientTests: XCTestCase {
             Thread.sleep(forTimeInterval: 0.01 + .random(in: -0.05...0.05))
         }
     }
+
+    func testAvoidLeakingTLSHandshakeCompletionPromise() {
+        let httpClient = HTTPClient(eventLoopGroupProvider: .createNew)
+        defer {
+            XCTAssertNoThrow(try httpClient.syncShutdown(requiresCleanClose: true))
+        }
+        XCTAssertThrowsError(try httpClient.get(url: "http://localhost/").wait()) { error in
```

Comment:
this test isn't great. It relies on nothing being bound to port 80 on `localhost`. Can I suggest a slightly different strategy: Let's bind a random port with a server socket, then immediately close it and then connect that port instead and we have pretty high confidence that this port won't be bound by something else

Reply:
@weissi Thank you! In https://github.com/swift-server/async-http-client/commit/cfba7e762d7faef23e42f66a438b538436ba973c, the  `refusesConnections: true` option is now set on `HTTPBin`, I think it gives the behaviour we want?

Reply:
Oops actually I don't think it does, I must still close it because the connection is still accepted it seems

Reply:
@weissi Latest push should be ok 🙂